sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -51,6 +51,93 @@ APIs for managing module identities, module twins, and querying modules
 
 ```csharp
 
+public  class Modules
+{
+    /// <summary>
+    /// Create a device module.
+    /// </summary>
+    /// <param name="moduleIdentity">The module to create.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns></returns>
+    public virtual async Task<Response<ModuleIdentity>> CreateIdentityAsync(ModuleIdentity moduleIdentity, CancellationToken cancellationToken = default);
```

Comment:
Can I assume that a moduleIdentity class will have the deviceId as a mandatory parameter for its constructor?

Reply:
That check will go in the implementation. and yes, it should have both the device id and the module Id